src/utils/calendar.ts

```diff
@@ -0,0 +1,34 @@
+import { PersonalizedVenueEvent } from "types/venues";
+import { VenueEvent } from "types/venues";
+import { eventEndTime, eventStartTime } from "utils/event";
+import { WithVenueId } from "utils/id";
+import { getFullVenueInsideUrl } from "utils/url";
+
+import ical from "ical-generator";
+
+export const createCalendar = (
+  allEvents: PersonalizedVenueEvent[] | WithVenueId<VenueEvent>[],
+  calendarname: string
+) => {
+  const cal = ical({ name: calendarname });
+
+  if (!cal) return;
+
+  allEvents.map((event: PersonalizedVenueEvent | WithVenueId<VenueEvent>) =>
```

Comment:
Since we're using the 'side effect' of processing these, rather than the return value, we should use `forEach` here:

I think we should also be able to leave off the explicit typing and just let TypeScript infer it:

```suggestion
  allEvents.forEach((event) =>
```

Further Reading:

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1423/commits/ec68e834fab42e9549a285d7a454e4c29e1f24e4